lib/replicate/checkpointer.js

```diff
@@ -93,8 +161,74 @@ Checkpointer.prototype.getCheckpoint = function () {
     if (err.status !== 404) {
       throw err;
     }
-    return 0;
+    return LOWEST_SEQ;
   });
 };
+// This checkpoint comparison is ported from CouchDBs source
+// they come from here:
+// https://github.com/apache/couchdb-couch-replicator/blob/master/src/couch_replicator.erl#L863-L906
+
+function compareReplicationLogs(srcDoc, tgtDoc) {
+  if(srcDoc.session_id === tgtDoc.session_id) {
+    return {
+      last_seq: srcDoc.last_seq,
```

Comment:
In checkpoint document root [`source_last_seq`](https://github.com/apache/couchdb-couch-replicator/blob/master/src/couch_replicator.erl#L871) field is used, not `last_seq`.